.github/workflows/release.yml

```diff
@@ -0,0 +1,52 @@
+name: envattrs release workflow
+
+on:
+  release:
+    types: [released] # only triggered on release, not pre-release or draft release
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  release:
+    name: Release to PyPI
+    runs-on: ubuntu-latest
+    env:
+      TARGET_BRANCH: ${{ github.event.release.target_commitish }}
+      DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
+      TAG_NAME: ${{ GITHUB_REF_NAME }}
+    steps:
+      - name: Check whether release target branch is default branch
+        if: ${{ TARGET_BRANCH != DEFAULT_BRANCH }}
+        uses: actions/github-script@v6
+        with:
+          script: |
+            core.setFailed('Target branch (${{ TARGET_BRANCH }}) is not default branch (${{ DEFAULT_BRANCH }}).')
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.7
+      - run: |
+          curl -sSL https://install.python-poetry.org | python -
+        env:
+          POETRY_VERSION: 1.3.1
+        shell: bash
+      - run: |
+          echo "PKG_VER=$(poetry version --short)" >> $GITHUB_ENV
+        shell: bash
+      - name: Check whether current tag matches package version
+        if: ${{ TAG_NAME != PKG_VER }}
+        uses: actions/github-script@v6
+        with:
+          script: |
+            core.setFailed('Tag name (${{ TAG_NAME }}) does not match package version (${{ PKG_VER }}).')
+      - name: Release
+        run: poetry publish --build
+        env:
+          POETRY_HTTP_BASIC_PYPI_USERNAME: ${{ secrets.PYPI_USERNAME }}
+          POETRY_HTTP_BASIC_PYPI_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
```

Comment:
can this not use api tokens? https://pypi.org/manage/account/token/

Reply:
Switched to token, scoped only to this project (envattrs).

Reply:
Added the token to keeper as well.